demo/ZeroToHeroUsingTheBasis.md

```diff
@@ -0,0 +1,357 @@
+sleep 5; \
+aca-py start \
+-it http '0.0.0.0' 10000 \
+-ot http \
+--admin '0.0.0.0' 5000 \
+-e 'http://localhost:10000/' \
+--auto-provision \
+--wallet-storage-type postgres_storage \
+--wallet-storage-config '{\"url\":\"host.docker.internal:5432\", \"wallet_scheme\":\"MultiWalletSingleTableSharedPool\"}' \
+--wallet-storage-creds '{\"account\":\"local-cloud-agent\",\"password\":\"testtest\",\"admin_account\":\"admin-cloud-agent\",\"admin_password\":\"testtest\"}' \
+--wallet-type indy \
+--wallet-name diwala \
+--wallet-key Diwala.Agent321311 \
+--seed diwalaJnecnKjegsyIR0P3vXBc56G71H \
+--genesis-url http://test.bcovrin.vonx.io/genesis \
+--label 'Diwala Multitenant Agent' \
+--admin-insecure-mode \
+--multitenant \
+--multitenant-admin \
+--jwt-secret very_secret_secret \
+--log-level info",
+]
+
+```
+
+## Postgres
+Postgres connection is important to make sure you are keeping state in a sane place. Volumes are fine, but they can be lost quite fast if it is not a good control over them. So a database, potentially as a service is even easier to handle and make sure that have the backups needed and overview needed.
+
+Setting up configuration for postgres is quite straight forward, with one hickup. The above configuration does not work out of the box, atleast not on < postgres@12.
+
+The setup fails because it cannot find the used “user” database. Which is weird, as the user is not a database, that is created during initialisation.
+
+To get past this, I had to create a “fake” database with the “user” name and it was able to run initialisation and all for the rest of the indy database creation.
+
+### Database management mode
+It is important to be aware of what is default and what possibilities you have for database management mode. Here is an explainer: https://github.com/hyperledger/indy-sdk/tree/master/experimental/plugins/postgres_storage#wallet-management-modes
+I chose MultiWalletSingleTableSharedPool because centralise all the things 🤪
+
+### Questions
+Is there any way to read out of the database and understand the binary data? Meaning setting up statistics towards the type column of a wallet db?
+When doing MultiWalletSingleTableSharedPool, why does it use the name wallets, and not the wallet name that the configuration sends in? There is an ID with the wallet name, but guess that is the multitenant wallet name one define then?
+
+# Creating connections
+Following this very straight forward: https://github.com/hyperledger/aries-cloudagent-python/blob/main/demo/AriesOpenAPIDemo.md#establishing-a-connection
+
+Important notice, one thing that was not straight forward in the demo and this setup was: Response does not go to active by default. There has to be an action on the connection to make it in active state.
+Trust ping or basic message pushes the state to active. But before that, it will stay response state.
```

Comment:
This is for the initial connection method based Aries RFC0160. For the new RFC0023 (DID Exchange) mechanism, this will not be needed as there is an extra message in the protocol to ensure the connection is confirmed.

Reply:
Info about this was added during this commit: https://github.com/hyperledger/aries-cloudagent-python/pull/1004/commits/284ac632e1dbb7810634fdcc2149755c4792dbca